core/core_test.go

```diff
@@ -127,7 +128,7 @@ func TestDefaultAPI_TokenRenew(t *testing.T) {
 				http.DefaultClient,
 				response.NoopValidator{},
 			)
-			_, err := api.TokenRenew("refresh me")
+			_, err := api.TokenRenew("refresh me", uuid.New())
```

Comment:
maybe you could create some tests to see that it is added into the requests header?
And that the value remains the same between requests, but changes at logout?